app/policies/project_test_policy.rb

```diff
@@ -27,12 +27,17 @@ def resolve
   end
 
   class Serializer < Serializer
+    include ScmHelper
+
     def to_builder options = {}
+      last_result = record.description.last_result
+      payload = options[:payloads].find{ |p| p.id == last_result.test_payload_id } if options[:payloads]
```

Comment:
The payload is pretty much required. `build_scm_data` will fail with an NPE if it isn't there. I would therefore check for the presence of `options[:payloads]` and `payload` here, and raise an error with a more useful message if they're not present, e.g. "Expected `options[:payloads]` to contain test payload with ID #{last_result.test_payload_id}"